app/src/main/java/org/dhis2/usescases/datasets/dataSetTable/dataSetSection/DataValuePresenter.java

```diff
@@ -513,8 +499,10 @@ public void initializeProcessor(@NonNull DataSetSectionFragment dataSetSectionFr
         for (int i = 0; i < listCategories.get(num).size(); i++) {
             if (num == 0)
                 current = new ArrayList<>();
-            if (current.size() == num + 1)
-                current.remove(current.size() - 1);
+            for (int j = current.size(); j > 0; j--) {
+                if (current.size() == num + j)
+                    current.remove(current.size() - j);
+            }
```

Comment:
What is "num", "i", "j" and "current" ? Can you give them a more expressive name ? ex: headerRowPositon etc..
What this expression (current.size() == num + j) means?

Reply:
Okey, I'll try to make it more expressive :sweat_smile:

Reply:
no problem :+1: